tests/pytests/test_mdis_drivers.py

```diff
@@ -34,199 +33,88 @@ def driver(request):
+        'image_lines': 512,
+        'image_samples': 512,
+        'name_platform': 'MESSENGER',
+        'name_sensor': 'MERCURY DUAL IMAGING SYSTEM NARROW ANGLE CAMERA',
+        'reference_height': {
+            'maxheight': 1000,
+            'minheight': -1000,
+            'unit': 'm'},
+        'name_model': 'USGS_ASTRO_FRAME_SENSOR_MODEL',
+        'center_ephemeris_time': 483122606.85252464}
+
+    label_file = get_image_label('EN1072174528M', label_type)
+
+    usgscsm_isd = ale.load(label_file, props={'kernels': test_kernels}, formatter='usgscsm')
+
+    assert compare_dicts(usgscsm_isd, isd) == []
+
+def test_spacecraft_name(driver):
+    assert driver.spacecraft_name == 'MESSENGER'
+
+def test_instrument_id(driver):
+    assert driver.instrument_id == 'MSGR_MDIS_NAC'
+
+def test_sampling_factor(driver):
+    assert driver.sampling_factor == 2
 
-def test_detector_start_line(driver):
-    assert driver.detector_start_line == 1
+def test_focal_length(driver):
+    with patch('ale.drivers.mes_drivers.spice.gdpool', return_value=[pow(4.07, -x) for x in np.arange(6)]) as gdpool, \
+         patch('ale.base.data_naif.spice.bods2c', return_value=-12345) as bods2c:
+         assert driver.focal_length == pytest.approx(6.0)
+         gdpool.assert_called_with('INS-12345_FL_TEMP_COEFFS', 0, 6)
 
 def test_detector_start_sample(driver):
-    assert driver.detector_start_sample == 9
-
-def test_sample_summing(driver):
-    assert driver.sample_summing == 2
-
-def test_line_summing(driver):
-    assert driver.line_summing == 2
-
-def test_platform_name(driver):
-    assert driver.platform_name.upper() == 'MESSENGER'
-
-def test_sensor_name(driver):
-    assert driver.sensor_name == 'MERCURY DUAL IMAGING SYSTEM NARROW ANGLE CAMERA'
+    with patch('ale.drivers.mes_drivers.spice.gdpool', return_value=[10.0]) as gdpool, \
+         patch('ale.base.data_naif.spice.bods2c', return_value=-12345) as bods2c:
+         assert driver.detector_start_sample == 10.0
+         gdpool.assert_called_with('INS-12345_FPUBIN_START_SAMPLE', 0, 1)
 
-def test_target_body_radii(driver):
-    np.testing.assert_equal(driver.target_body_radii, [2439.4, 2439.4, 2439.4])
+def test_detector_start_line(driver):
+    with patch('ale.drivers.mes_drivers.spice.gdpool', return_value=[10.0]) as gdpool, \
+         patch('ale.base.data_naif.spice.bods2c', return_value=-12345) as bods2c:
+         assert driver.detector_start_line == 10.0
+         gdpool.assert_called_with('INS-12345_FPUBIN_START_LINE', 0, 1)
 
-def test_focal_length(driver):
-    assert driver.focal_length == 549.5535053027719
+def test_detector_center_sample(driver):
+    with patch('ale.drivers.mes_drivers.spice.gdpool', return_value=[512.5, 512.5, 1]) as gdpool, \
+         patch('ale.base.data_naif.spice.bods2c', return_value=-12345) as bods2c:
+         assert driver.detector_center_sample == 512
```

Comment:
This is a SPICE call for the ISIS driver, but not for the PDS3 driver. So, I'm not checking if gdpool was called here.